packages/api/src/app-router.ts

```diff
@@ -143,6 +157,25 @@ export default async function makeApp(params: CliArgs) {
     req.stripe = stripe;
     next();
   });
+  app.use(authenticator());
+  app.use(
+    cors(
+      corsOptsProvider({
+        anyOriginPathPrefixes: [pathJoin("/", httpPrefix, "/asset/upload/")],
+        jwtOrigin: corsJwtAllowlist,
+        baseOpts: {
+          credentials: true,
+          exposedHeaders: ["*"],
+        },
+      })
+    )
+  );
```

Comment:
There is currently 1 problem with this middleware order:

If an invalid token or JWT is used for the request, the `authenticator` middleware will explode and then the `cors` middleware will never run. This means that if you use an invalid token from the browser you might never even see the response, because the response won't have CORS headers and then the browser will explode the request in a different way for you. 

To fix this, we have to make the `authenticator` middleware never throw an exception. It can be made either by actually never exploding and just letting the `authorizer` middleware explode later with "request not authenticated", or we can do some kind of deferred explode, saving the error in the request to throw after the `cors` middleware. Orr we can also make the CORS an error handler somehow?

I'll explore these possibilites tomorrow.

Reply:
Fixed it here! 
https://github.com/livepeer/livepeer-com/pull/985/commits/89d1fac7299d04b9d9e79600976ed7ba798bce6e